packages/components/src/snackbar/README.md

```diff
@@ -2,15 +2,117 @@
+3. [Development guidelines](#development-guidelines)
+4. [Related components](#related-components)
 
 ## Design guidelines
 
-A Snackbar displays a succinct message that is cleared out after a small delay. It can also offer the user options, like viewing a published post but these options should also be available elsewhere in the UI.
+### Usage
+
+A Snackbar displays a temporary, succinct message. It can also offer the user the option to perform a single action, like view a published post. 
+
+#### Principles
+
+- **Informational:** Snackbars provide updates on a process. Use them to let users know that things are working as expected.
+- **Temporary:** Snackbars appear temporarily, and disappear on their own.
+- **Contextual:** Snackbars appear in the context in which the user is working, and don’t get in the way.
+
+#### Actions
+
+A Snackbar can optionally contain a single action. Snackbars disappear automatically, whether the user interacts with them or not, so that action should not be “Dismiss” or “Cancel.” The action should also be available elsewhere in the UI.
+
+#### When to Use
+
+Snackbars communicate low-priority messages that are minimally interruptive and don’t require user action.
+
+For error messages, or for important, persistent messages, use a Notice instead.
+
+### Anatomy
+
+![](https://wordpress.org/gutenberg/files/2019/06/snackbar-anatomy.png)
+
+1. Text label
+2. Container
+3. Action (Optional)
+4. Focus state
+
+#### Text Label
+
+Snackbars contain a text label that directly relates to the process being described.
+Text labels are short, clear updates, and normally display on one line. 
+
+![](https://wordpress.org/gutenberg/files/2019/06/snackbar-text-label-do1.png)
+
+**Do:** Let labels wrap onto two lines on small screens.
+
+![](https://wordpress.org/gutenberg/files/2019/06/snackbar-text-label-do2.png)
+**Do:** Let the width of Snackbars expand on larger screens.
+
+#### Containers
+
+Snackbars are displayed in rectangular containers with a dark grey background. Containers should be completely opaque so the text labels remain legible. 
+
+In wide layouts, the container width can be extended to accommodate longer text labels.
+
+![](https://wordpress.org/gutenberg/files/2019/06/snackbar-container-dont1.png)
+
+**Don’t:** Do not change the color of the background.
+
+![](https://wordpress.org/gutenberg/files/2019/06/snackbar-container-dont2.png)
+
+**Don’t:** Do not use icons within a Snackbar.
+
+### Placement
+
+Snackbars appear on the bottom left of the screen. On smaller screens, they can take up the available width. 
+
+### Action
+
+Along with their text labels, Snackbars can display a single text button to let users interact with the process being performed. Snackbars shouldn’t be the only way to access a core function or to make something usable.
+
+![](https://wordpress.org/gutenberg/files/2019/06/snackbar-action-dont1.png)
+
+**Don’t:** Do not use buttons inside of snackbars. 
```

Comment:
this also falls in the same area as the other comment, class names should not be available to customization because the button should not be customized 
> A `className` property can be used to add custom classes to the button styles.